### ISSUE TITLE
Custom Dataset

### ISSUE THREAD

**Issue by QasimKhan5x** [CoinCheung/BiSeNet#93]
I am using this dataset for [human parsing](https://www.kaggle.com/yuqizhu/human-parsing-dataset)
However, I made the segmentation masks into binary masks such that black pixels correspond to the background and white pixels correspond to humans, which means labels are grayscale. Also, images are .jpg but labels are .png. I am training it on the dataset but getting this
![image](https://user-images.githubusercontent.com/29240497/97701937-fec7ee00-1acf-11eb-9c38-a1a46b0d5a4f.png)
Can you guide plz?

Here is a reference:

## Image
![image](https://user-images.githubusercontent.com/29240497/97703027-bb6e7f00-1ad1-11eb-85f3-d1e834a196e3.png)
## Label
![image](https://user-images.githubusercontent.com/29240497/97703058-c9240480-1ad1-11eb-9055-80ae74a47721.png)

**Comment by CoinCheung** [comment]
Hi, 
What modifications have you made, and how many classes you have set to your model? Have you ever tried to use normal cross-entropy-loss(or bce loss) instead of ohem loss ?

**Comment by QasimKhan5x** [comment]
> Hi,
> What modifications have you made, and how many classes you have set to your model? Have you ever tried to use normal cross-entropy-loss(or bce loss) instead of ohem loss ?

Suppose I just want two classes - one corresponding to the background and the other corresponding to human. How can I set it in the model? Do I just have to change loss to BCE?

**Comment by CoinCheung** [comment]
you need to change the number of model output classes. If you want to use bce, you need to change to 1 class, and if you want to use softmax-cross-entropy, you need to change it to 2 classes. And you need to check how many classes in your label files.

**Comment by QasimKhan5x** [comment]
My label files are just these grayscale images with white pixels for human and black pixels for anything else. So I think I just have one class which corresponds to human

**Comment by Eashwar93** [comment]
@QasimKhan5x could you tell me how exactly you solved it?

1. What is the loss function that you used?
2. What is your cityscapes_cv2.py look like?
3. How many classes did you create your network with?

It would be really nice if you could share me these 3 details. I also have a 2 class problem very similar to yours. I used the following:
1. OHEM CE loss
2. `train-id "1" for background` and `traind-id "2" for monorail`, and `lb_ignore=255`. Monorail refers to the object that I am interested in.

> code snippets from cityscapes_cv2.py

```
labels_info = [
    {"hasInstances": False, "category": "void", "catid": 0, "name": "background", "ignoreInEval": False, "id": 0, "color": [0, 0, 0], "trainId": 1},
    {"hasInstances": True, "category": "void", "catid": 1, "name": "monorail", "ignoreInEval": False, "id": 1, "color": [128, 64,128], "trainId": 2},
]
```
```
self.n_cats = 2
self.lb_ignore = 255
```
3. I created network with 2 classes in my train_amp.py

code snippets from tain_amp.py

```
def set_model():
    net = model_factory[cfg.model_type](2)
```

I also verified my labeling and I think there is no mistake regarding the same as well.

**Comment by Eashwar93** [comment]
This issue exists only when I train using AMP. The normal training completes successfully. When I make use of AMP., the forward pass results in Nan resulting in the loss to be Nan. If I find a fix to use AMP with Custom dataset, will comeback here and respost
